wagtail/admin/tests/tests.py

```diff
@@ -333,15 +333,36 @@ def test_no_permissions(self):
 class Test404(TestCase, WagtailTestUtils):
     def test_admin_404_template_used(self):
```

Comment:
Suggestion: If I'm understanding this correctly, it looks like we are only testing the 404 view using the default settings (`APPEND_SLASH = True`).

However, depending on the value of `APPEND_SLASH` we use a slightly different urlconfig. Can we add a test case for `APPEND_SLASH = False` as well? Even though it seems rather insignificant it adds to the test coverage. See my comment below on mocking settings for more information on how to do this.

Reply:
Thanks for the feedback @Stormheg!
So trying to do this, I faced the same issue I mentioned in the following [comment](https://github.com/wagtail/wagtail/issues/5814#issuecomment-830854643).
I will put the [link to Django docs](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#use-of-django-conf-settings) which explains why:

> Modules should not in general use settings stored in django.conf.settings at the top level 

And that's what we are doing here; which makes mocking settings harder.

So I tried to write a test but it fails - even tough we have the expected behaviour when running a Wagtail project.
I inspected the `tried` argument provided by ResolverMatch and found that the URL_CONF is set with APPEND_SLASH=True.
I forced the APPEND_SLASH setting to be False in my local  django and the test passes.

Should we rewrite it considering django's advice to use:

> a level of laziness or indirection ..., such as django.utils.functional.LazyObject, django.utils.functional.lazy() or lambda.?

Or is there a way i can rewrite the test ?

Reply:
Thank you for your explanation @Tijani-Dia. I did not consider that when writing the suggestion. I'm afraid I don't know of an any easy way to work around this issue so I'd be happy to accept it as-is.

Of course you're free to try but I don't think it's worth the effort. Not sure how other core team members feel about this though - calling for @gasman's expert advice here :wink:. Maybe reloading the urlconfig during the test to make sure `settings.APPEND_SLASH` is reevaluated might work? But that sounds like quite a hassle...

Reply:
Alright! I made the changes you suggested, waiting for the expert's advice!

Reply:
The only non-hacky way I can think of is to add a separate test run for the APPEND_SLASH=False setup, a bit like we do for [testing custom user models and USE_TZ=False](https://github.com/wagtail/wagtail/blob/9d29ba862c94a81d6cddb4010b179c6fce450799/.github/workflows/test.yml#L20-L32), and that seems like overkill here. I agree, it's not worth the hassle - I'm OK with leaving the APPEND_SLASH=False case untested.

Reply:
Thank you both for your input! Marking this as resolved :heavy_check_mark: